### PR TITLE
Updates to match bootstrapped plugin

### PR DESCRIPTION
**Pull request by vegaro**
The purpose of this PR is to update our project to match the structure of a bootstrapped plugin. A fair amount of things have changed in React Native since we created the plugin, so this should bring the latest development tools:

- The example iOS project now uses CocoaPods
- The example Android project has now a task `enableLocalBuild`, same way we do in Flutter. This way we can specify our local installation of purchases-hybrid-common and be able to edit, the common files and the plugin files when opening the example Android project with Android Studio
- There is a new yarn script (yarn bootstrap) that effectively does the same as our previous `setup.example`. This was part of the plugin created from scratch with the react native tool. I thought it looked cleaner and it's better to use whatever is the standard.
- The CircleCI config file has also been simplified by using the official Orb. It also has improvements regarding caching.
- There's a tool https://www.npmjs.com/package/release-it that came in the from-scratch plugin. It is very interesting and we should explore it, but I haven't included it in this PR.

### CODE REVIEW

File: __tests__/index.test.js [hunk 1]
```diff
@@ -484,7 +483,7 @@ describe("Purchases", () => {
   it("syncpurchases works for android", () => {
     const Purchases = require("../index").default;
 
-    mockPlatform("android");
+    Platform.OS = "android";
```
Comment (vegaro) [inline __tests__/index.test.js:486]: This has changed in the more recent version of RN

File: android/build.gradle [hunk 2]
```diff
@@ -1,40 +1,128 @@
...
-    compileSdkVersion safeExtGet("compileSdkVersion", 28)
-    buildToolsVersion safeExtGet("buildToolsVersion", "28.0.3")
+    compileSdkVersion getExtOrIntegerDefault('compileSdkVersion')
+    buildToolsVersion getExtOrDefault('buildToolsVersion')
 
     defaultConfig {
-        minSdkVersion safeExtGet('minSdkVersion', 16)
-        targetSdkVersion safeExtGet('targetSdkVersion', 26)
+        minSdkVersion 16
+        targetSdkVersion getExtOrIntegerDefault('targetSdkVersion')
```
Comment (vegaro) [inline android/build.gradle:31]: This has been increased from 26 to 28, but our Android plugin is already targeting 28, so it should be safe

File: example/android/app/build.gradle [hunk 3]
```diff
@@ -176,3 +206,38 @@ task copyDownloadableDepsToLibs(type: Copy) {
...
+  doLast {
+    configurations.findAll().each { config ->
+      if (config.name.contains("minReactNative") && config.canBeResolved) {
+        print config.name
+        print '\n'
+        config.files
+      }
+    }
+  }
+}
```
Comment (vegaro) [inline example/android/app/build.gradle:243]: This task is required by the CircleCI orb

File: example/android/app/src/debug/java/com/revenuecat/purchases_sample/ReactNativeFlipper.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+                      }
+                    });
+              }
+            });
+      } else {
+        client.addPlugin(new FrescoFlipperPlugin());
+      }
+    }
+  }
+}
```
Comment (vegaro) [inline example/android/app/src/debug/java/com/revenuecat/purchases_sample/ReactNativeFlipper.java:69]: This is for Flipper (https://fbflipper.com). It also comes by default. It's very useful actually, and better to have to use chrome for debugging. It's basically the react native debugger.

File: example/android/settings.gradle [hunk 5]
```diff
@@ -1,7 +1,15 @@
...
+
 include ':react-native-purchases'
 project(':react-native-purchases').projectDir = new File(rootProject.projectDir, '../../android')
 
-include ':app'
+// Run enableLocalBuild task to enable building purchases-hybrid-common from your local copy
+if (file(".composite-enable").exists()) {
+    String path = new File(".composite-enable").text
+    includeBuild (path)
+}
```
Comment (vegaro) [inline example/android/settings.gradle:15]: This is what adds the local purchases-hybrid-common

File: example/ios/Podfile [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+  pod 'React-jsinspector', :path => '../node_modules/react-native/ReactCommon/jsinspector'
+  pod 'ReactCommon/callinvoker', :path => "../node_modules/react-native/ReactCommon"
+  pod 'ReactCommon/turbomodule/core', :path => "../node_modules/react-native/ReactCommon"
+  pod 'Yoga', :path => '../node_modules/react-native/ReactCommon/yoga', :modular_headers => true
+
+  pod 'DoubleConversion', :podspec => '../node_modules/react-native/third-party-podspecs/DoubleConversion.podspec'
+  pod 'glog', :podspec => '../node_modules/react-native/third-party-podspecs/glog.podspec'
+  pod 'Folly', :podspec => '../node_modules/react-native/third-party-podspecs/Folly.podspec'
+
+  pod 'RNPurchases', :path => '../..'
```
Comment (vegaro) [inline example/ios/Podfile:56]: This is the default Podfile too

File: example/ios/ReactNativeSample-tvOS/Info.plist [hunk 7]
```diff
@@ -1,54 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
```
Comment (vegaro) [inline example/ios/ReactNativeSample-tvOS/Info.plist:1]: The default plugin doesn't have this target

File: example/package.json [hunk 8]
```diff
@@ -20,20 +22,21 @@
...
   "devDependencies": {
-    "@babel/core": "^7.5.5",
-    "babel-core": "^7.0.0-0",
+    "@babel/core": "^7.9.6",
+    "@babel/runtime": "^7.9.6",
+    "babel-plugin-module-resolver": "^4.0.0",
+    "metro-react-native-babel-preset": "^0.59.0",
     "babel-jest": "24.9.0",
     "jest": "24.9.0",
-    "metro-react-native-babel-preset": "0.56.0",
```
Comment (vegaro) [inline example/package.json:36]: Increased the versions to the ones in the default plugin

File: setupJest.js [hunk 9]
```diff
@@ -714,4 +703,7 @@ NativeModules.RNPurchases = {
   setPushToken: jest.fn(),
 };
 
-jest.mock('NativeEventEmitter');
+jest.mock(
+  'react-native/Libraries/EventEmitter/NativeEventEmitter',
+ );
```
Comment (vegaro) [inline setupJest.js:708]: The mock is now provided by RN

File: .npmignore [hunk 10]
```diff
@@ -1,14 +0,0 @@
...
-local.properties
-.jest
-.vscode
-src/
-tsconfig.json
-*.iml
-android/build
-.DS_Store
-.circleci/config.yml
-RELEASING.md
```
Comment (vegaro) [inline .npmignore:14]: this moved to package.json

File: package.json [hunk 11]
```diff
@@ -5,16 +5,39 @@
...
+    "!ios/RNPurchases.xcodeproj/project.xcworkspace",
+    "!ios/RNPurchases.xcodeproj/xcuserdata",
+    "RNPurchases.podspec",
+    "scripts",
+    "!**/__tests__",
+    "!**/.idea",
+    "!**/*.iml",
+    "!**/.DS_Store",
+    "!**/.gitignore"
+  ],
```
Comment (vegaro) [inline package.json:30]: I ran `npm publish` before adding all this, and it was trying to publish 13.5 MB. After these changes:

```
npm notice 📦  react-native-purchases@3.3.2
npm notice === Tarball Contents ===
npm notice 2.5kB  package.json
npm notice 616B   build.js
npm notice 8.8kB  CHANGELOG.md
npm notice 32.6kB index.d.ts
npm notice 27.7kB index.js
npm notice 1.1kB  LICENSE
npm notice 4.5kB  README.md
npm notice 773B   RNPurchases.podspec
npm notice 3.8kB  android/build.gradle
npm notice 150B   android/gradle.properties
npm notice 0      android/settings.gradle
npm notice 137B   android/src/main/AndroidManifest.xml
npm notice 4.1kB  android/src/main/java/com/revenuecat/purchases/react/RNPurchasesConverters.kt
npm notice 9.9kB  android/src/main/java/com/revenuecat/purchases/react/RNPurchasesModule.java
npm notice 963B   android/src/main/java/com/revenuecat/purchases/react/RNPurchasesPackage.java
npm notice 360B   ios/RNPurchases.h
npm notice 10.6kB ios/RNPurchases.m
npm notice 9.8kB  ios/RNPurchases.xcodeproj/project.pbxproj
npm notice 2.6kB  ios/strip-frameworks.sh
npm notice 1.3kB  scripts/download-purchases-common.sh
npm notice 1.2kB  scripts/download-purchases-framework.sh
npm notice === Tarball Details ===
npm notice name:          react-native-purchases
npm notice version:       3.3.2
npm notice package size:  26.7 kB
npm notice unpacked size: 123.4 kB
```

**Review by aboedo — APPROVED** [review]
looking great so far!

File: example/ios/Podfile.lock [hunk 12]
```diff
@@ -0,0 +1,468 @@
+PODS:
```
[the remark this answers is not in the document]
Comment (vegaro) [reply example/ios/Podfile.lock]: We actually gain faster CI builds since we are using checksum of Podfile.lock as a cache key to cache the Pods folder

**Pull request merged**